CSharp/Microsoft.Bot.Builder.Teams/Middlewares/TeamsTenantFilteringMiddleware.cs

```diff
@@ -51,23 +51,22 @@ public TeamsTenantFilteringMiddleware(IEnumerable<string> allowedTenantIds)
         /// </remarks>
         /// <seealso cref="ITurnContext" />
         /// <seealso cref="Schema.IActivity" />
-        public Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default(CancellationToken))
         {
-            // Ignoring cases where ChannelData is missing or does not contain TenantId.
-            if (turnContext.Activity.ChannelData != null)
+            TeamsChannelData teamsChannelData = turnContext.Activity.GetChannelData<TeamsChannelData>();
```

Comment:
I am not comfortable with us dropping every single message without tenant Id in it. This makes it hard for people to test using other channels. Let's split the middleware into 2 parts. One which drops non MsTeams activities and one which denies MsTeams Activities if the tenant Id is wrong or missing. We should not be blocking other channels.

Reply:
Here is my reasoning; If I as a user have decided to use the teams tenant filtering middleware... then I expect ONLY messages from a tenant to flow through... since a tenant is only identified in the teams channel... then I implicitly have also decided to only allow messages from the teams channel.

The middleware isn't mandatory and so if people don't want to block tenants then they won't use it. What do you think?

Reply:
It is not uncommon for people to test bot locally using emulator or something else and then eventually deploying to run against MsTeams. We should not assume the use case. We have a clear way of blocking messages and allow only MsTeams and then do tenant filtering. This ensures we don't block messages of other channels for no reasons.

People might want to block MsTeams messages unless they are from specific tenant but at the same time allow messages from other channels to flow through. It is not a very fat fetched idea. Think of an org transitioning from Slack to MsTeams. I am sure there will be countless other scenarios.

Reply:
But then conversely, if I do apply this middleware wouldn't I expect all messages to be filtered to my tenant? Having messages still flow through from outside of my chosen tenants would be rather unexpected. Anyway, I don't have too strong of an opinion on this I can make it only check tenant for the msteams channel.

Reply:
Documentation will need to help with this sadly. But I would rather not shut down doors.

Reply:
I still think it would be clearer if it just blocked all messages but I updated it to what you wanted and added some tests